lib/.eslintrc.js

```diff
@@ -18,6 +18,8 @@ module.exports = {
   },
   rules: {
     "node/no-unsupported-features/es-syntax": 0,
+    "@typescript-eslint/no-non-null-assertion": 0,
+    "no-use-before-define": 0,
```

Comment:
are there many instances of this? we are currently disabling the rule on a case by case basis since this could be hiding an error. particularly the non-null assertion could be better handled with some code refactor, so I see it as mostly a debt item to refactor the code to have early return or type guard functions so that TS knows whether a variable can be null or not

Reply:
good call, will take a closer look

Reply:
removed, and replaced with comment in code